### PR TITLE
Cut as_lisp_obj

### PR DESCRIPTION
**Pull request by nickdrozd**
Prefer into where possible, or else LispObject::from

### CODE REVIEW

**Comment by nickdrozd** [comment]
This takes care of all of them except for `LispObarrayRef`, which does some borrowing stuff that I haven't been able to figure out.

File: rust_src/src/chartable.rs [hunk 1]
```diff
@@ -235,11 +235,7 @@ pub fn set_char_table_parent(mut chartable: LispCharTableRef, parent: Option<Lis
         }
     }
 
-    chartable.parent = if let Some(p) = parent {
-        p.as_lisp_obj()
-    } else {
-        Qnil
-    };
+    chartable.parent = if let Some(p) = parent { p.into() } else { Qnil };
```
Comment (shaleh) [inline rust_src/src/chartable.rs]:
```suggestion
    chartable.parent = parent.map_or(Qnil, |p| p.into());
```

I think the `map_or` makes more sense. But if not, wouldn't `match` be a better fit? Not on you, this was existing code. I think we over used the `if let Some()` idiom.

**Comment by shaleh** [comment]
I do not see the issue with `LispObarrayRef`. It is a new type wrapper around `LispObject` like `LispCons`.

**Comment by shaleh** [comment]
Can you also remove the `LispCons` `as_obj`?

**Comment by nickdrozd** [comment]
I think that should take care of all of them.

File: rust_src/src/chartable.rs (hunk 1 above)
Comment (nickdrozd) [reply rust_src/src/chartable.rs]: We can do even better than that: `parent: Option<LispCharTableRef>`, so all we need is `chartable.parent = parent.into()`.

**Pull request merged**